src/module/actor/lancer-actor.ts

```diff
@@ -1110,6 +1169,53 @@ export class LancerActor extends Actor {
   is_deployable(): this is LancerActor & { data: LancerActorDataProperties<EntryType.DEPLOYABLE> } {
     return this.data.type === EntryType.DEPLOYABLE;
   }
+
+
+  /**
+   * Taking a new and old frame/class, swaps the actor and/or token images if 
+   * we detect that the image isn't custom. Will check each individually
+   * @param robot     A MM Mech or NPC, passed through to avoid data overwrites 
+   * @param oldFrame  Old Frame or NPC Class
+   * @param newFrame  New Frame or NPC Class
+   * @returns         The newFrame if any updates were performed
+   */
+  async swapFrameImage(robot: Mech | Npc, oldFrame: Frame | NpcClass | null, newFrame: Frame | NpcClass): Promise<string> {
+    let oldFramePath = frameToPath[oldFrame?.Name || ""];
+    let newFramePath = frameToPath[newFrame?.Name || ""];
+    let defaultImg = robot.Type == EntryType.MECH ? "systems/lancer/assets/icons/mech.svg" : "systems/lancer/assets/icons/npc_class.svg";
+    
+    if(!newFramePath) newFramePath = defaultImg;
+    let changed = false;
+    let newData: any = {}
```

Comment:
This can be `Parameters<this["update"]>[0]` instead of `any` to get better type checking.

Reply:
Ooh very nice